shared/login/login/index.js

```diff
@@ -75,7 +80,8 @@ export default connect(
       onForgotPassphrase: () => dispatch(openAccountResetPage()),
       onLogin: (user, passphrase, store) => dispatch(relogin(user, passphrase, store)),
       onSignup: () => dispatch(routeAppend(['signup'])),
-      onSomeoneElse: () => { dispatch(login()) }
+      onSomeoneElse: () => { dispatch(login()) },
+      onSaveInKeychainChange: (user, saveInKeychain) => { if (!saveInKeychain) dispatch(clearStoredSecret(user)) }
```

Comment:
nvm looks good either way, this might be a little simpler since the component doesn't understand about `clearStoredSecret` and we can get rid of the component state some time in the future and replace it with an action.